.github/workflows/java_thin_client_compatibility.yaml

```diff
@@ -0,0 +1,117 @@
+name: Test Java Thin client against the released IMDG servers
+
+on:
+  workflow_dispatch:
+    inputs:
+      organization_name:
+        description: Default is hazelcast, but if you would like to run the workflow with your forked repo, set your github username
+        required: true
+        default: hazelcast
+      branch_name:
+        description: Name of the branch to test client from
+        required: true
+        default: master
+
+jobs:
+  setup_server_matrix:
+    name: Setup the server test matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Checkout to scripts
+        uses: actions/checkout@v2
+
+      - name: Set server matrix
+        id: set-matrix
+        run: |
+          versions=$( python get_server_matrix.py )
+          echo "Versions from Python script: $versions"
+          filtered_versions=$(echo "$versions" | jq -c '[.[] | select(. > "5.2")]')
+          echo "Filtered versions: $filtered_versions"
+          echo "matrix=$filtered_versions" >> "$GITHUB_OUTPUT"
+
+  test_client:
+    needs: [setup_server_matrix]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        server-version: ${{ fromJson(needs.setup_server_matrix.outputs.matrix) }}
+        kind: [os, enterprise]
+        os: [ ubuntu-latest ]
+        java: [ { version: 17, distribution: corretto },
+                { version: 21, distribution: corretto },
+                { version: 17, distribution: zulu },
+                { version: 21, distribution: zulu },
+                { version: 17, distribution: semeru },
+                { version: 21, distribution: semeru },
+                { version: 17, distribution: oracle },
+                { version: 21, distribution: oracle },
+                { version: 17, distribution: microsoft },
+                { version: 21, distribution: microsoft }
+              ]
+
+    name: Test Java Thin client ${{ github.event.inputs.branch_name }} branch against ${{ matrix.kind }} ${{ matrix.server-version }} server on ${{ matrix.os }} and JVM ${{ matrix.java.version }} ${{ matrix.java.distribution }}
+    steps:
+      - name: Checkout to scripts
+        uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ matrix.java.version }}
+          distribution: ${{ matrix.java.distribution }}
+
+      - name: Checkout to ${{ github.event.inputs.branch_name }}
+        uses: actions/checkout@v2
```

Comment:
does checkout@v4 work here? If so, can you use it? Let's use the same version in this file.

Reply:
I will fix this

Reply:
Fixed